File: python/algorithm/KMP/KMP_test.py
from kmp import cal_next, KMP


def test_cal_next():
    next = cal_next('ababaca')
    assert next == [-1, -1, 0, 1, 2, -1, 0]


def test_kmp():
    assert 4 == KMP('安保处abcdefg', 'bcde')

if __name__ == '__main__':
    test_cal_next()
    test_kmp()

File: go/algorithm/KMP/KMP_test.go
package KMP

import (
    "testing"
    "fmt"
)

func TestCalNext(t *testing.T) {
    s := "ababaca"
    fmt.Println(calNext(s))
}

func TestGetNext(t *testing.T) {
    s := "ababaca"
    fmt.Println(getNext(s))
}

func TestKMP(t *testing.T) {
    src, desc := "测abcdefg", "bcde"
    startIndex := KMP(src, desc)
    startIndexB := KMP_BASIC(src, desc)
    if 4 != startIndex || startIndex != startIndexB {
        t.Error("KMP error: result: ", startIndex, startIndexB)
    }
}
